src/cli/run.rs

```diff
@@ -85,14 +91,106 @@ pub async fn run<P: AsRef<Path>>(
     )?;
 
     if task_outcome == TaskOutcome::Failure {
+        stop_notify.notify_one();
+        let _ = inquiry_handle.await;
         return Ok(());
     }
 
     squash_merge_branch(path, &base_branch, &fork_branch)?;
-
+    stop_notify.notify_one();
+    let _ = inquiry_handle.await;
     Ok(())
 }
 
+async fn handle_inquiries(agent_base_url: String, agent_api_key: String, stop_notify: Arc<Notify>) {
+    let client = reqwest::Client::new();
+
+    loop {
+        // Check stop signal
+        let stop_future = stop_notify.notified();
+        let sleep_future = tokio::time::sleep(std::time::Duration::from_secs(1));
+        tokio::select! {
+            _ = stop_future => {
+                println!("[handle_inquiries] Received stop_notify. Shutting down handler!");
+                break;
+            }
+            _ = sleep_future => {
+                let url = format!("{agent_base_url}/api/agent/inquiry_request");
```

Comment:
Okay, I see your using the `/inquiry_request` and `/inquiry_response` endpoints to communicate within the CLI.
Please find another way to directly get the request/set the response in the `InquiryState`.
Do we even need a separate handler thread? With the current design of the `/inquiry` endpoint, you can just call this logic in the handler.

I don't think we want to keep an endpoint around that blocks waiting for user input, but it's okay for an initial version.